src/util/metrics/threads_linux.rs

```diff
@@ -61,23 +69,41 @@ impl Collector for ThreadsCollector {
     }
 
     fn collect(&self) -> Vec<proto::MetricFamily> {
-        let cpu_totals = self.cpu_totals.lock().unwrap();
-        let tids = get_thread_ids(self.pid).unwrap();
+        // Synchronous collecting.
```

Comment:
do we run the collect in multi-threads?

Reply:
No, currently we collect metrics in a single-threading `promepusher`.